src/plugins/saved_objects_management/common/types/README.md

```diff
@@ -0,0 +1,39 @@
+## Versioned interfaces
+
+This folder contains types that are shared between the server and client:
+
+```ts
+// v1.ts
+export interface SavedObjectWithMetadata { name: string }
+```
+
+This type is _versioned_ because it lives in the file `v1.ts`. Thus it is "v1"
+of this type.
+
+**Do not alter a versioned type**. Types may be in use by clients. Alterations
+must be made on a new version of the TS interface. Versions are determined
+using monotonically increasing numbers: 1, 2, 3, etc.
```

Comment:
Folks using the SO management plugin changes as a reference might interpret the statement as "all internal versioned interfaces have to be denoted as monotonically increasing numbers." I don't think we intend to proscribe any versioning implementation here, so maybe we could rephrase this a bit to say:
```
Versions in this plugin are determined using monotonically increasing numbers: 1, 2, 3, etc.
```